tests/expand-manifest.jsonld

```diff
@@ -0,0 +1,326 @@
+      "input": "expand/st18-in.jsonld",
+      "expect": "expand/st18-out.jsonld",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst19",
+      "@type": ["jld:PositiveEvaluationTest", "jld:ExpandTest"],
+      "name": "Annotation node 2",
+      "purpose": "Node with @annotation property on node object",
+      "input": "expand/st19-in.jsonld",
+      "expect": "expand/st19-out.jsonld",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst20",
+      "@type": ["jld:PositiveEvaluationTest", "jld:ExpandTest"],
+      "name": "Annotation node 3",
+      "purpose": "Node with @annotation property multiple values",
+      "input": "expand/st20-in.jsonld",
+      "expect": "expand/st20-out.jsonld",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst21",
+      "@type": ["jld:NegativeEvaluationTest", "jld:ExpandTest"],
+      "name": "Invalid annotation node 1",
+      "purpose": "Node with @annotation property that is on the top-level is invalid",
+      "input": "expand/st21-in.jsonld",
+      "expectErrorCode": "invalid annotation",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst22",
+      "@type": ["jld:NegativeEvaluationTest", "jld:ExpandTest"],
+      "name": "Invalid annotation node 2",
+      "purpose": "Node with @annotation property on a top-level graph node is invalid",
+      "input": "expand/st22-in.jsonld",
+      "expectErrorCode": "invalid annotation",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst23",
+      "@type": ["jld:NegativeEvaluationTest", "jld:ExpandTest"],
+      "name": "Invalid annotation node 3",
+      "purpose": "Node with @annotation property having @id is invalid",
+      "input": "expand/st23-in.jsonld",
+      "expectErrorCode": "invalid annotation",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst24",
+      "@type": ["jld:NegativeEvaluationTest", "jld:ExpandTest"],
+      "name": "Invalid annotation node 4",
+      "purpose": "Node with @annotation property with value object value is invalid",
+      "input": "expand/st24-in.jsonld",
+      "expectErrorCode": "invalid annotation",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
+    }, {
+      "@id": "#tst25",
+      "@type": ["jld:NegativeEvaluationTest", "jld:ExpandTest"],
+      "name": "Invalid annotation node 5",
+      "purpose": "Node with @annotation on a list",
+      "input": "expand/st25-in.jsonld",
+      "expectErrorCode": "invalid annotation",
+      "requires": "JSON-LD*",
+      "option": {"specVersion": "json-ld*", "rdfstar": true}
```

Comment:
Well, I would not necessarily forbid that...
Wouldn't it make sense to annotate `<< ex:bob ex:knows _:b >>` here,
where `_:b` is the first bnode of the list?

By the way, the current text in the spec does not forbid it, AFAICT.

Reply:
You can annotate something that is the value of `rdf:rest`, but we don't support it on values of `@list`. Turtle doesn't support this in any way, either.

From Grammar extensions:

> A node object which is not the value of a non-keyword property, or which is an value of a list object, MUST NOT contain an @annotation property.

Reply:
(I realize that, regardless of its location, this `@annotation` is invalid because it is a string. I overlooked it, and I suggest we change it into something that would otherwise be valid. I will make that change)

In this test, it is not a value of the list object that contains `@annotation`, it is the list object itself. So my understanding is that it is not forbidden by this text.

This test is, in my view, roughly equivalent of the following (valid) Turtle*:
```
ex:bob ex:knows ( ex:fred ) {| ex:some ex:annotation |}.
```
resulting in
```
ex:bob ex:knows _:b0.
_:b0 rdf:first ex:fred.
_:b0 rdf:rest rdf:nil.
<< ex:bob ex:knows _:b0 >> ex:some ex:annotation.
```
and therefore it would be nice to also support it in JSON-LD. NB: this would require updating the definition of "list object" (which allows "an `@index` key, but no other entries").

Writing that, I realize that this creates another asymmetry between `@list` and `@set`, which some users might find confusing...

Reply:
Well, there's currently a restriction that `@annotation` can only be used on node objects and value objects. To annotate a list, it would be on a list object. We could certainly extend to do that, but it wouldn't work for terms which are `@container` `@list` (as we don't for straight string values). There is some work in the node map algorithm to figure this out, but shouldn't be a show-stopper.

Reply:
> Well, there's currently a restriction that `@annotation` can only be used on node objects and value objects.

Silly me! :-/ You are right, of course.

> but it wouldn't work for terms which are `@container @list`

No, of course. But similarly, you cannot annotate an edge if you use a native JSON value (e.g. 42, "hello" or false) as its object -- you need to introduce a value object.

Reply:
We'd need to restrict the recursive list case, though. Consider:

`{"v": [["b"]]}` which might be annotated as follows:

```
{
  "v": {
    "@list": [
      {"@list": "b", "@annotation": {"p": "o"}
    ]
  }
}
```

You couldn't do this in Turtle*, but limiting this in JSON-LD* might be a bit more difficult algorithmically. We could only support the case where the list object as a direct value of some other property, and not an embedded list.

Reply:
Would'nt that fall under "the value of a list object MUST NOT contain an `@annotation` property"? Granted, it is not covered at the moment because the sentence is about "node objects". But the sentence needs fixing anyway, because this restriction currently must apply also to value objects --I just noticed that ;-)